client/signup/steps/emails/test/index.tsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { render } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import * as useQueryProductsList from 'calypso/components/data/query-products-list';
+import EmailSignupTitanCard from '../index.jsx';
+
+const initialState = {
+	flowName: 'test:flow',
+	stepName: 'test:step2',
+	sites: {
+		items: {
+			1: {},
+		},
+	},
+	ui: {},
+	productsList: {
+		isFetching: false,
+	},
+};
+
+const productListQuerySpy = jest.spyOn( useQueryProductsList, 'default' );
+
+describe( 'Email Step Titan Signup Card', () => {
+	test( 'should request a complete product list', () => {
+		const newInitialState = {
+			...initialState,
+			sites: {
+				...initialState.sites,
+				items: {
+					1: {
+						...initialState.sites.items[ 1 ],
+					},
+				},
+			},
+		};
+		const middlewares = [ thunk ];
+		const mockStore = configureStore( middlewares );
+		const store = mockStore( newInitialState );
+
+		render(
+			<Provider store={ store }>
+				<EmailSignupTitanCard { ...newInitialState } />
+			</Provider>
+		);
+
+		expect( productListQuerySpy ).toHaveBeenCalled();
+		// An empty first argument means we're fetching all products, so we'll
+		// have access to email product data.
+		expect( productListQuerySpy.mock.calls[ 0 ][ 0 ] ).toStrictEqual( {} );
```

Comment:
```suggestion
		expect( productListQuerySpy ).toHaveBeenCalledWith( {}, {} );
```
I played around to see if it was possible to spy directly on `useQueryProductsList`, but it doesn't look like it is (likely because it's defined **and** called locally in `client/components/data/query-products-list/index.jsx`).

So, we're really spying on the React component here–which is fine–but it means that the function is receiving a second object argument (the React children, I believe). 

Anyway, it would be nice to collapse these two assertions into a single `toHaveBeenCalledWith`. The assertion you're using still follows Jest's documented APIs, so this is no blocker—it just seems a bit more elegant.

Reply:
I played around with that too, but ultimately had the same result.

I just did a quick local test and confirmed that yes, that second prop is the React children. I'll update to a single `toHaveBeenCalledWith` assertion.